### PR TITLE
[MNOE-1296] App mgmt requirements

### PR DESCRIPTION
**Pull request by adamaziz15**
- Disable ordering if billing details are required
- Disable all app management if billing details or payment are required
- Add disclaimer confirm modal when saving billing details
- Add toastr on successful update of billing details
- Update mnoe config svc with app management requirement settings

screen shot 2018-11-11 at 11 05 07 pm
screen shot 2018-11-11 at 11 05 40 pm

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -27,7 +27,7 @@
     "eslint-config-xo-space": "^0.15.0",
     "eslint-plugin-angular": "^1.3.0",
     "glob": "^7.1.2",
-    "gulp": "gulpjs/gulp#4.0",
+    "gulp": "^4.0.0",
```
Comment (ouranos) [inline package.json]: :+1:

File: src/app/components/mno-app-install-btn/mno-app-install-btn.coffee [hunk 2]
```diff
@@ -69,13 +69,17 @@ angular.module 'mnoEnterpriseAngular'
...
       vm.updateButtonDisabledTooltip = () ->
-        if !vm.canProvisionApp
+        if vm.billingDetailsRequired
+          'mno_enterprise.templates.components.app_install_btn.billing_details_req'
+        else if !vm.canProvisionApp
           'mno_enterprise.templates.components.app_install_btn.insufficient_privilege'
         else if !vm.orderPossible
           'mno_enterprise.templates.dashboard.marketplace.show.no_pricing_plans_found_tooltip'
+        else if vm.conflictingApp
+          $translate.instant('mno_enterprise.templates.components.app_install_btn.conflicting_app') + ' ' + vm.conflictingApp.name
```
Comment (ouranos) [inline src/app/components/mno-app-install-btn/mno-app-install-btn.coffee]: Why is this one using a `$translate.instant`, could you just use interpolation in the locale string?

File: src/app/components/mno-apps/mno-app.coffee [hunk 3]
```diff
@@ -87,6 +87,14 @@ angular.module 'mnoEnterpriseAngular'
 
       MnoeOrganizations.get().then((response) -> vm.user_role = response.current_user.role)
 
+      # Is an up to date account required to allow app management and is the account past due?
```
Comment (ouranos) [inline src/app/components/mno-apps/mno-app.coffee]: This feels like déjà vu :stuck_out_tongue_closed_eyes: 
Time for a helper?

File: src/locales/en.json [hunk 4]
```diff
@@ -623,6 +623,11 @@
   "mno_enterprise.templates.dashboard.organization.credit_card.billing_form.postcode": "Postcode",
   "mno_enterprise.templates.dashboard.organization.credit_card.billing_form.country": "Country",
   "mno_enterprise.templates.dashboard.organization.credit_card.billing_form.country_select": "-- SELECT --",
+  "mno_enterprise.templates.dashboard.organization.credit_card.confirm_modal.cancel": "Cancel",
+  "mno_enterprise.templates.dashboard.organization.credit_card.confirm_modal.confirm": "Ok",
+  "mno_enterprise.templates.dashboard.organization.credit_card.confirm_modal.header": "<b>Payment terms</b>",
+  "mno_enterprise.templates.dashboard.organization.credit_card.confirm_modal.disclaimer": "<span class='text-danger'>Please note: This credit card will be used to process all outstanding invoices automatically on the 2nd of each month. Click ok to continue.</span>",
```
Comment (ouranos) [inline src/locales/en.json]: The span tag should be in the template not in the locale

Comment (adamaziz15) [reply src/locales/en.json]: Updated the component to allow dynamic classes and removed span tags from the locale. 👍

File: src/locales/en.json [hunk 5]
```diff
@@ -623,6 +623,11 @@
   "mno_enterprise.templates.dashboard.organization.credit_card.billing_form.postcode": "Postcode",
   "mno_enterprise.templates.dashboard.organization.credit_card.billing_form.country": "Country",
   "mno_enterprise.templates.dashboard.organization.credit_card.billing_form.country_select": "-- SELECT --",
+  "mno_enterprise.templates.dashboard.organization.credit_card.confirm_modal.cancel": "Cancel",
+  "mno_enterprise.templates.dashboard.organization.credit_card.confirm_modal.confirm": "Ok",
+  "mno_enterprise.templates.dashboard.organization.credit_card.confirm_modal.header": "<b>Payment terms</b>",
```
[the remark this answers is not in the document]
Comment (adamaziz15) [reply src/locales/en.json]: Updated to remove those tags 👌

File: src/app/services/org-account-helper/org-account-helper.coffee [hunk 6]
```diff
@@ -13,7 +13,7 @@ angular.module 'mnoEnterpriseAngular'
       # Is an up to date account required to allow app management and is the account past due?
       paymentRequired = MnoeConfig.isCurrentAccountRequired() && selectedOrg.organization.in_arrears
       # Are billing details required and are they present?
-      detailsRequired = MnoeConfig.areBillingDetailsRequired() && _.isEmpty(selectedOrg.credit_card)
+      detailsRequired = MnoeConfig.areBillingDetailsRequired() && !selectedOrg.credit_card.presence
```
Comment (ouranos) [inline src/app/services/org-account-helper/org-account-helper.coffee]: Does that work in JS?
Or is this a field exposed by the backend?

Comment (adamaziz15) [reply src/app/services/org-account-helper/org-account-helper.coffee]: It's a field exposed by the backend. I saw this was done in the admin panel and ran into an issue where the credit card isn't returned in the response for a member or manager user. So I copied the pattern here. For member or manager users the credit card response only contains this attribute.

Comment (adamaziz15) [reply src/app/services/org-account-helper/org-account-helper.coffee]: Updated to only show id attr of credit cards when user is a member or manager.

**Pull request merged**